commit f697554515b06e8d7264f316b25e6da943407142
Author: Aurelien Jarno <aurelien@aurel32.net>
Date:   Fri May 2 17:02:23 2008 +0200

    KVM: PIT: support mode 3
    
    The in-kernel PIT emulation ignores pending timers if operating
    under mode 3, which for example Hurd uses.
    
    This mode should output a square wave, high for (N+1)/2 counts and low
    for (N-1)/2 counts. As we only care about the resulting interrupts, the
    period is N, and mode 3 is the same as mode 2 with regard to
    interrupts.
    
    Signed-off-by: Aurelien Jarno <aurelien@aurel32.net>
    Signed-off-by: Avi Kivity <avi@qumranet.com>

diff --git a/arch/x86/kvm/i8254.c b/arch/x86/kvm/i8254.c
index 735ec9a..60074dc 100644
--- a/arch/x86/kvm/i8254.c
+++ b/arch/x86/kvm/i8254.c
@@ -308,6 +308,7 @@ static void pit_load_count(struct kvm *kvm, int channel, u32 val)
 		create_pit_timer(&ps->pit_timer, val, 0);
 		break;
 	case 2:
+	case 3:
 		create_pit_timer(&ps->pit_timer, val, 1);
 		break;
 	default:
